### ISSUE TITLE
Question about the implementation of the confidence-based loss

### ISSUE THREAD

**Issue by Vison307** [Dootmaan/ICMIL#2]
First of all, thank you for the interesting work!

However, when I read the code, I could not find the implementation of the confidence score $\sigma(a(x))$ in Eq. (8) of the paper.

$$\mathcal{L} = \sigma(a(x)) \cdot \mathcal{L}_c + \alpha \mathcal{L}_w$$

For example, in `step2_teacher_student_abmil_distillation.py`, the loss is implemented in [L139](https://github.com/Dootmaan/ICMIL/blob/c2fd2faa4e85cf85527e228573b434a8cc461cba/step2_teacher_student_abmil_distillation.py#L139C1-L140C1):

```python
loss0 = ce_cri(F.log_softmax(tPredict_student,dim=-1), F.softmax(tPredict_teacher.detach(),dim=-1)).mean()
loss1 = ce_cri(F.log_softmax(consistency_tPredict,dim=-1), F.softmax(tPredict_teacher,dim=-1)).mean()
loss2 = ce_cri(F.log_softmax(consistency_tmidFeat,dim=-1), F.softmax(tmidFeat_teacher,dim=-1)).mean()
loss=loss0+0.5*loss1+0.5*loss2
```

I assume `loss1` and `loss2` are $\mathcal{L}_w$ in Eq. (8), and `loss0` is $\mathcal{L}_c$, but there is no $\sigma(a(x))$. Have I missed something?

**Comment by Dootmaan** [comment]
Hi @Vison307! Thank you for your question! This is actually a very good question so I have tagged it as a "good first issue." 😊

This project has two branches, of which the "main" branch holds the code for our MICCAI paper and the "confidence_based" branch holds the code for our TMI journal version. In the "confidence_based" branch, the confidence-based ICMIL method is reflected in the step2_train_teacherstudent_[method]\_ax_semilearn.py files. To work with the confidence-based method, please switch to the confidence_based branch and refer to the step2_train_teacherstudent_[method]_ax_semilearn.py files. These scripts are specifically designed for the confidence-based training workflow.

We truly appreciate you bringing this up! Apologies for the current file structure—it’s not ideal at the moment. We’ll review and improve the organization to make it more user-friendly.

**Comment by Vison307** [comment]
Thank you very much for your kind and prompt reply! I find the code now!
